controllers/cloudstackmachine_controller.go

```diff
@@ -67,35 +68,39 @@ type CloudStackMachineReconciler struct {
 func NewCSMachineReconciliationRunner() *CloudStackMachineReconciliationRunner {
 	// Set concrete type and init pointers.
 	r := &CloudStackMachineReconciliationRunner{ReconciliationSubject: &infrav1.CloudStackMachine{}}
-	r.CAPIMachine = &capiv1.Machine{}
+	r.CAPIMachine = &clusterv1.Machine{}
 	r.StateChecker = &infrav1.CloudStackMachineStateChecker{}
-	r.Zones = &infrav1.CloudStackZoneList{}
 	r.IsoNet = &infrav1.CloudStackIsolatedNetwork{}
 	r.AffinityGroup = &infrav1.CloudStackAffinityGroup{}
-	r.FailureDomain = &infrav1.CloudStackZone{}
+	r.FailureDomain = &infrav1.CloudStackFailureDomain{}
 	// Setup the base runner. Initializes pointers and links reconciliation methods.
-	r.ReconciliationRunner = utils.NewRunner(r, r.ReconciliationSubject)
+	r.ReconciliationRunner = utils.NewRunner(r, r.ReconciliationSubject, "CloudStackMachine")
 	return r
 }
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (reconciler *CloudStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, retErr error) {
-	return NewCSMachineReconciliationRunner().
-		UsingBaseReconciler(reconciler.ReconcilerBase).
-		ForRequest(req).
-		WithRequestCtx(ctx).
-		RunBaseReconciliationStages()
+	r := NewCSMachineReconciliationRunner()
+	r.UsingBaseReconciler(reconciler.ReconcilerBase).ForRequest(req).WithRequestCtx(ctx)
+	r.WithAdditionalCommonStages(
+		r.GetParent(r.ReconciliationSubject, r.CAPIMachine),
+		r.RequeueIfCloudStackClusterNotReady,
+		r.SetFailureDomainOnCSMachine,
+		r.GetObjectByName("placeholder", r.FailureDomain,
+			func() string { return r.ReconciliationSubject.Spec.FailureDomainName }),
+		r.CheckPresent(map[string]client.Object{"CloudStackFailureDomain": r.FailureDomain}),
+		r.AsFailureDomainUser(&r.FailureDomain.Spec))
+	return r.RunBaseReconciliationStages()
 }
 
 func (r *CloudStackMachineReconciliationRunner) Reconcile() (retRes ctrl.Result, reterr error) {
 	return r.RunReconciliationStages(
-		r.GetZonesAndRequeueIfMissing(r.Zones),
-		r.GetParent(r.ReconciliationSubject, r.CAPIMachine),
-		r.RequeueIfCloudStackClusterNotReady,
+		r.GetObjectByName("placeholder", r.IsoNet,
+			func() string { return r.IsoNetMetaName(r.FailureDomain.Spec.Zone.Network.Name) }),
+		r.RunIf(func() bool { return r.FailureDomain.Spec.Zone.Network.Type == cloud.NetworkTypeIsolated },
+			r.CheckPresent(map[string]client.Object{"CloudStackIsolatedNetwork": r.IsoNet})),
```

Comment:
does this mean we only run the cloudstackmachinereconciliation if we're using an isolated network?

Reply:
No, the `RunIf` method runs `CheckPresent` if the network type is an isolated network.

It's just a quick way to require an isolated network was found if we need one.